### PR TITLE
Bump Scala to 2.12.x and SBT to 1.2.x

### PR DESCRIPTION
**Pull request by adelbertc**
Bumped the versions and the rest followed..

### CODE REVIEW

File: core/src/main/scala/Actionable.scala [hunk 1]
```diff
@@ -71,15 +71,14 @@ object Actionable {
...
-        val unitw = Manifest.Versioned.unwrap(unit)
         val ttl = cfg.cleanup.initialTTL.toSeconds
         val exp = Instant.now.plusSeconds(ttl)
         val plan = actionConfig.plan
         val dc = actionConfig.datacenter
         val ns = actionConfig.namespace
         val hash = actionConfig.hash
         val policy =
-          if (Manifest.isPeriodic(unitw, plan))
+          if (Manifest.isPeriodic(plan))
```
Comment (timperrett) [inline core/src/main/scala/Actionable.scala]: Why is the unit not being passed in here?

Comment (adelbertc) [reply core/src/main/scala/Actionable.scala]: It just wasn't being used in the `isPeriodic` definition - all the changes in this PR besides the version bumps are I guess new scalac flags being turned on and triggering errors with unused variables/imports.

**Comment by adelbertc** [comment]
Am I clear to merge?

**Comment by timperrett** [comment]
@adelbertc how have you tested it? this is a big change, so it would be good to get some validation

**Comment by adelbertc** [comment]
Is there much to test here though given it's just bump the Scala version and follow the warnings? No feature was added/removed, just stuff like unused imports and variables and dead code.

**Comment by timperrett** [comment]
I guess we just bumped to 0.14, so i'm ok I guess. These things just make me nervous 🤣 Do you need a release with this @adelbertc ?

**Comment by adelbertc** [comment]
😆 🙏 

No need for a release, this was step 1 of me working on the control plane bits.

**Pull request merged**